### PR TITLE
fix: avoid name clashes with generated variable names

### PR DESCRIPTION
**Pull request by mlutze**
Fixes https://github.com/flix/flix/issues/2681

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 1]
```diff
@@ -682,7 +682,7 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
                   // If the tag is `Some` we construct the lambda: x -> Some(x).
 
                   // Construct a fresh symbol for the formal parameter.
-                  val freshVar = Symbol.freshVarSym("x", BoundBy.FormalParam, loc)
+                  val freshVar = Symbol.freshVarSym(Flix.Delimiter + "x", BoundBy.FormalParam, loc)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:685]: I am worried about names starting with the delimiter. I would go for `tmp$delim` or whatever.

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:685]: How about ending with the delimiter? It looks like that's what we do in the Weeder.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala]: Thats fine

**Comment by magnus-madsen** [comment]
(I guess we should also ban `%` in a Flix name -- if it is not already).

**Pull request merged**